addon/webextension/selector/shooter.js

```diff
@@ -196,11 +196,25 @@ this.shooter = (function() { // eslint-disable-line no-unused-vars
     }));
   };
 
+  let copyInProgress = null;
   exports.copyShot = function(selectedPos) {
+    // This is pretty slow. We'll ignore additional user triggered copy events
+    // while it is in progress.
+    if (copyInProgress) {
+      return;
+    }
+    // A max of five seconds in case some error occurs.
+    copyInProgress = setTimeout(() => {
+      copyInProgress = null;
+    }, 5000);
     let dataUrl = screenshotPage(selectedPos);
     let blob = blobConverters.dataUrlToBlob(dataUrl);
     catcher.watchPromise(callBackground("copyShotToClipboard", blob).then(() => {
       uicontrol.deactivate();
+      if (copyInProgress) {
+        clearTimeout(copyInProgress);
+        copyInProgress = null;
+      }
```

Comment:
If this project rejects then it should also clear the timeout (though generally that shouldn't happen). If `Promise.finally` was available that's what should really get used. I don't know if we have a good module to put a finally implementation into (there's one in test.js).

Maybe also add `disabled` to the button while it's in progress? I think we should probably add a different icon while it is in progress, but that applies to all the buttons so we should do that as a separate bug (plus we need an icon).